compiler-rt/lib/mpk_untrusted/mpk_fault_handler.cpp

```diff
@@ -53,22 +53,22 @@ void disablePageMPK(siginfo_t *si, void *arg) {
 void disableThreadMPK(void *arg, uint32_t pkey) {
   uint32_t *pkru_ptr = __mpk_untrusted::pkru_ptr(arg);
 
-  last_pkey = pkey;
-  last_access_rights = __mpk_untrusted::pkey_get(pkru_ptr, pkey);
+  auto handler = __mpk_untrusted::AllocSiteHandler::get();
+  auto pkey_info = __mpk_untrusted::PKeyInfo(
+      pkey, __mpk_untrusted::pkey_get(pkru_ptr, pkey));
+  handler->storePidKey(getpid(), pkey_info);
   __mpk_untrusted::pkey_set(pkru_ptr, pkey, PKEY_ENABLE_ACCESS);
 
   __sanitizer::Report("INFO : Pkey(%d) has been set to ENABLE_ACCESS to enable "
                       "instruction access.\n",
                       pkey);
 }
 
-void enableThreadMPK(void *arg, uint32_t pkey) {
+void enableThreadMPK(void *arg, __mpk_untrusted::PKeyInfo pkey_info) {
```

Comment:
Just a general comment. since you're basically operating within a private namespace and in a cpp file, it is fine to use a `using` declaration to avoid what are normally unambiguous namespace prefixes. It would also be perfectly valid to put your code within the` __mpk_untrusted` namespace. `using` declarations are almost always OK in implementations, but should pretty much never happen in headers. being explicit is fine, but not strictly necessary.

Reply:
Added this portion of code to the `__mpk_untrusted` namespace.